### PR TITLE
 Make Asset#type a getter that offers a guess based on Content-Type, incoming relations, file extension etc.

### PR DESCRIPTION
**Pull request by papandreou**
As discussed on the gitter channel. Was easier than I expected.

I've copy/pasted some code from Asset#loadAsync, and there's still a chance to dedupe it some. LMK if it seems worthwhile to proceed :)

### CODE REVIEW

File: lib/assets/Asset.js [hunk 1]
```diff
@@ -114,6 +114,39 @@ class Asset extends EventEmitter {
...
+                return typeFromExtension;
+            } else if (commonTargetType) {
+                return commonTargetType;
+            }
+        }
+        return this._type;
+    }
+
+    set type(type) {
+        this._type = type;
```
Comment (Munter) [inline lib/assets/Asset.js]: Should there be a setter at all, and if so, should this not include the logic to detect sub typing or switches to different asset type family branches?

File: test/transforms/populate.js [hunk 2]
```diff
@@ -14,7 +14,8 @@ describe('transforms/populate', function () {
                 followRelations: {type: query.not('HtmlStyle')}
             });
 
-        expect(assetGraph, 'to contain no assets', 'Css');
+        expect(assetGraph, 'to contain asset', 'Css');
+        expect(assetGraph, 'to contain asset', { type: 'Css', isLoaded: false });
```
Comment (Munter) [inline test/transforms/populate.js]: This seems to just be a more specific version of the previous assertion. Kill the less specific version?

**Comment by Munter** [comment]
It's great to see this improving the quality of the graph from your test adjustments :D

File: test/transforms/populate.js (hunk 2 above)
Comment (papandreou) [reply test/transforms/populate.js]: You're right, fixed! It was late :)

File: lib/assets/Asset.js (hunk 1 above)
Comment (papandreou) [reply lib/assets/Asset.js]: I assumed that there would be some places where the `type` property would be set by `createAsset` or friends, but apparently I didn't bother to check. It works fine without the setter, and I've removed it now.

**Pull request merged**